lib/src/content/components/main_content/wolt_modal_sheet_main_content.dart

```diff
@@ -48,53 +137,77 @@ class WoltModalSheetMainContent extends StatelessWidget {
     final isNonScrollingPage = page is NonScrollingWoltModalSheetPage;
     final shouldFillRemaining = woltModalType.forceMaxHeight ||
         (page.forceMaxHeight && !isNonScrollingPage);
-    final scrollView = CustomScrollView(
-      shrinkWrap: true,
-      physics: themeData?.mainContentScrollPhysics ??
-          defaultThemeData.mainContentScrollPhysics,
-      controller: scrollController,
-      slivers: [
-        if (!isNonScrollingPage)
-          SliverList(
-            delegate: SliverChildBuilderDelegate(
-              (context, index) {
-                if (index == 0) {
-                  final heroImage = page.heroImage;
-                  return heroImage != null
-                      ? WoltModalSheetHeroImage(
-                          topBarHeight: topBarHeight,
-                          heroImage: heroImage,
-                          heroImageHeight: heroImageHeight,
-                          scrollAnimationStyle: scrollAnimationStyle,
-                        )
-                      // If top bar layer is always visible, the padding is explicitly added to the
-                      // scroll view since top bar will not be integrated to scroll view at all.
-                      // Otherwise, we implicitly create a spacing as a part of the scroll view.
-                      : SizedBox(
-                          height:
-                              isTopBarLayerAlwaysVisible ? 0 : topBarHeight);
-                } else {
-                  final pageTitle = page.pageTitle;
-                  return KeyedSubtree(
-                    key: pageTitleKey,
-                    child: pageTitle ?? const SizedBox.shrink(),
-                  );
+
+    final scrollView = NotificationListener(
+      onNotification: _shouldDragContent
```

Comment:
instead of this you can extract the main content widget to a variable `mainContent`

```dart
if (_shouldDragContent) {
   return NotificationListener(..., child: mainContent)
}

return mainContent;
```